pkg/datapath/linux/config/config.go

```diff
@@ -390,6 +391,17 @@ func (h *HeaderfileWriter) WriteNodeConfig(w io.Writer, cfg *datapath.LocalNodeC
 		cDefinesMap["NATIVE_DEV_MAC_BY_IFINDEX(IFINDEX)"] = macByIfIndexMacro
 		cDefinesMap["IS_L3_DEV(ifindex)"] = isL3DevMacro
 	}
+
+	// Enabling NetfilterCompatibleMode increases BPF instruction count size, and may
+	// cause issues in kernels that have lower instruction complexity limit. Hence enabling,
+	// this feature only in kernels support higher instruction complexity limit.
+	// NetfilterCompatibleMode feature can be enabled on all kernels, once we have support
+	// for adding new tail calls in host device[bpf_host].
+	if option.Config.NetfilterCompatibleMode && probes.NewProbeManager().GetMisc().HaveLargeInsnLimit &&
```

Comment:
We already checked kernel support during the agent bootstrap in `daemon/cmd/daemon_main.go`, so why do we need to check again here?

Reply:
I don't recall why this check was added, looks like redundant. Updated PR.